Sources/Logging/Logging.swift

```diff
@@ -505,30 +505,64 @@ public struct MultiplexLogHandler: LogHandler {
     }
 }
 
-/// Ships with the logging module, really boring just prints something using the `print` function
-internal struct StdoutLogHandler: LogHandler {
-    private let lock = Lock()
+/// A wrapper to facilitate `print`-ing to stderr and stdio that
+/// ensures access to the underlying `FILE` is locked to prevent
+/// cross-thread interleaving of output.
+public struct StdioOutputStream: TextOutputStream {
```

Comment:
The `StdioOutputStream` we should leave `internal` or `private` as that's a thing the stdlib should really provide.

Reply:
Yeah, I have a similar feeling but keeping it private requires some kind of additional wrapper type or switch to power `StreamLogHandler`'s public initializer (fundamentally providing the consumer the ability to switch between stderr and stdout). 

As I had it before `StreamLogHandler.Stream` also conformed to `TextOutputStream`.

Previously it was
```
public static let stderr: Stream = .init(underlying: FileOutputStream(file: systemStderr))
```

now this has the same level of access:
```
public static let stdout = StdioOutputStream(file: systemStdout)
```

but with less indirection.

Ultimately _something_ has to be public in that initializer because one of the goals of this change is to allow easily redirected to `stderr`. Alternatively, to keep the type private, I could have multiple specialized initializers or possibly static factory methods to provide this functionality.

Reply:
@allenhumphreys if you overload the `init` of `StreamLogHandler` instead of using default arguments, it'll work. All the streams stuff can stay `internal`/`private`. So

```swift
public struct StreamLogHandler { 
    ...
    public init(label: String) {
        self.init(stream: StdioOutputStream(stream: sysStdout))
    }

    public init(label: String, stream: TextOutputStream) {
        ...
    }
    ...
}
```

Reply:
That doesn't leave a good name for an init that goes to `stderr`, any suggestion good or not as good?
Just thinking out loud:
```
init(toStdoutWithLabel label: String)
init(toStderrWithLabel label: String)
```

Reply:
@allenhumphreys Ahh, I see! I'd then go with

```swift
public struct StreamLogHandler {
    public static func makeStdoutLogHandler(label: String) -> StreamLogHandler { ... }
    public static func makeStderrLogHandler(label: String) -> StreamLogHandler { ... }

   ...
}
```

and then we can even leave the `init` private